### PR TITLE
fix: not allowed to specify `create_time` and `update_time` when create/edit route, service, upstream and consumer

### LINKED ISSUE #933: feat:  not allowed to specify `create_time` and `update_time` when create/edit route, service, upstream and consumer
```go
{
        caseDesc: "update the route (r1)",
        Object:   ManagerApiExpect(t),
        Path:     "/apisix/admin/routes/r1",
        Method:   http.MethodPut,
        Body: `{
                        "uri": "/hello",
                        "name": "new jack",
                        "create_time": 1606726340,    // throw an error if it was specified
                        "update_time": 1606726340,    // throw an error if it was specified
                        "desc": "new desc",
                        "upstream": {
                                "type": "roundrobin",
                                "nodes": [{
                                        "host": "172.16.238.20",
                                        "port": 1980,
                                        "weight": 1
                                }]
                        }
                }`
},
```

original comment: https://github.com/apache/apisix-dashboard/pull/927/files#r533114051

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
close #933

### CODE REVIEW

File: api/filter/schema.go [hunk 1]
```diff
@@ -106,6 +106,18 @@ func parseCert(crt, key string) ([]string, error) {
 }
 
 func handleSpecialField(resource string, reqBody []byte) ([]byte, error) {
+	var bodyMap map[string]interface{}
+	err := json.Unmarshal(reqBody, &bodyMap)
+	if err != nil {
+		return reqBody, fmt.Errorf("read request body failed: %s", err)
+	}
+	if _, ok := bodyMap["create_time"]; ok {
+		return reqBody, fmt.Errorf("not support specifying create_time")
```
Comment (juzhiyuan) [inline api/filter/schema.go:115]: We may have a better error message, e.g `We don't accept create_time from client`?

File: api/test/e2e/route_with_management_fileds_test.go [hunk 2]
```diff
@@ -363,3 +363,75 @@ func TestRoute_search_by_label(t *testing.T) {
...
+			ExpectStatus: http.StatusBadRequest,
+		},
+		{
+			caseDesc: "create route with update_time",
+			Object:   ManagerApiExpect(t),
+			Path:     "/apisix/admin/routes/r1",
+			Method:   http.MethodPut,
+			Body: `{
+				"uri": "/hello",
+				"update_time": 1,
```
Comment (juzhiyuan) [inline api/test/e2e/route_with_management_fileds_test.go:394]: 🤣 not a good timestamp though it works.

Comment (nic-chen) [reply api/test/e2e/route_with_management_fileds_test.go]: updated.

File: api/filter/schema.go (hunk 1 above)
Comment (nic-chen) [reply api/filter/schema.go]: updated.

File: api/filter/schema.go [hunk 3]
```diff
@@ -106,6 +106,18 @@ func parseCert(crt, key string) ([]string, error) {
 }
 
 func handleSpecialField(resource string, reqBody []byte) ([]byte, error) {
+	var bodyMap map[string]interface{}
+	err := json.Unmarshal(reqBody, &bodyMap)
+	if err != nil {
+		return reqBody, fmt.Errorf("read request body failed: %s", err)
+	}
+	if _, ok := bodyMap["create_time"]; ok {
+		return reqBody, fmt.Errorf("we don't accept create_time from client")
```
Comment (tokers) [inline api/filter/schema.go:115]: Here we don't add any parameters, just use errors.New is fine.

File: api/filter/schema.go [hunk 4]
```diff
@@ -106,6 +106,18 @@ func parseCert(crt, key string) ([]string, error) {
...
+	}
+	if _, ok := bodyMap["update_time"]; ok {
+		return reqBody, fmt.Errorf("we don't accept update_time from client")
```
Comment (tokers) [inline api/filter/schema.go:118]: Ditto.

File: api/filter/schema.go (hunk 3 above)
Comment (gxthrj) [reply api/filter/schema.go:115]: If need to format your error message using `fmt.Errorf()`.
If we do not need format using `errors.New()`.

Comment (nic-chen) [reply api/filter/schema.go]: fixed.

File: api/filter/schema.go (hunk 4 above)
Comment (nic-chen) [reply api/filter/schema.go]: fixed.

**Review by juzhiyuan — APPROVED** [review]
LGTM for Route's test cases, should we add some test cases for Service, Upstream, and consumer, too?

**Review by tokers — APPROVED** [review]

**Pull request merged**